compose/cli/main.py

```diff
@@ -565,17 +565,25 @@ def up(self, project, options):
         start_deps = not options['--no-deps']
         service_names = options['SERVICE']
         timeout = int(options.get('--timeout') or DEFAULT_TIMEOUT)
+        detached = options.get('-d')
 
         to_attach = project.up(
             service_names=service_names,
             start_deps=start_deps,
             strategy=convergence_strategy_from_opts(options),
             do_build=not options['--no-build'],
-            timeout=timeout
+            timeout=timeout,
+            detached=detached
         )
 
-        if not options['-d']:
-            log_printer = build_log_printer(to_attach, service_names, monochrome)
+        if not detached:
+            filtered_streams = [(container, thread) for container, thread in to_attach if thread is not None]
+            containers, log_streams = zip(*filtered_streams)
+            log_printer = LogPrinter(
+                list(containers),
+                monochrome=monochrome,
+                log_streams=list(log_streams)
+            )
```

Comment:
It looks like you removed `build_log_printer` but then reproduced most of the logic inline and didn't remove the function?

Could this be moved back into `build_log_printer` ?